Examples/Io/Root/src/RootBFieldWriter.cpp

```diff
@@ -140,27 +142,22 @@ void RootBFieldWriter::run(const Config& config,
         nBinsY = nBins.at(0);
         nBinsZ = nBins.at(1);
       } else {
-        std::ostringstream errorMsg;
-        errorMsg << "BField has wrong dimension. The dimension needs to be "
-                    "either 2 (r,z,Br,Bz) or 3(x,y,z,Bx,By,Bz) in order to be "
-                    "written out by this writer.";
-        throw std::invalid_argument(errorMsg.str());
+        throw std::invalid_argument(
+            "BField has wrong dimension. The dimension needs to be "
+            "either 2 (r,z,Br,Bz) or 3(x,y,z,Bx,By,Bz) in order to be "
+            "written out by this writer.");
       }
     }
 
-    stepX = fabs(minX - maxX) / nBinsX;
-    stepY = fabs(minY - maxY) / nBinsY;
-    stepZ = fabs(minZ - maxZ) / nBinsZ;
+    stepX = fabs(minX - maxX) / (nBinsX - 1);
+    stepY = fabs(minY - maxY) / (nBinsY - 1);
+    stepZ = fabs(minZ - maxZ) / (nBinsZ - 1);
```

Comment:
should we switch to `std::abs` here?

Reply:
Good question, which led me down a rabbit hole.
 * I don't know which is better in practice, but I prefer `std::abs`.
 * both `fabs` and `std::abs` should have `#include <cmath>`.
 * If we reversed `min` and `max`, we might not need `abs` at all. If the user specifies them the wrong way round, then the subsequent code wouldn't work with `fabs`, but might still work without. So maybe it's better to remove these `fabs` altogether.

```suggestion
    stepX = (maxX - minX) / (nBinsX - 1);
    stepY = (maxY - minY) / (nBinsY - 1);
    stepZ = (maxZ - minZ) / (nBinsZ - 1);
```

What do you think?

Reply:
I agree we should be able to assume that it is ordered. maybe we can add an assertion for max > min or exception as you prefer!

Reply:
Good idea! I changed it.